src/cert_context.rs

```diff
@@ -107,6 +121,69 @@ impl CertContext {
         self.get_encoded_bytes()
     }
 
+    /// Deletes the cert's associated key container for both CAPI and CNG keys
+    pub fn delete_key_container(&mut self) -> io::Result<()> {
+        unsafe {
+            // get CERT_KEY_PROV_INFO_PROP_ID and cast to key_prov
+            let bytes = self.get_bytes(Cryptography::CERT_KEY_PROV_INFO_PROP_ID)?;
+            assert!(bytes.len() <= u32::max_value() as usize);
+            let key_prov = bytes.as_ptr() as *const Cryptography::CRYPT_KEY_PROV_INFO;
```

Comment:
I wonder if it is worth having a `KeyProv` (`KeyProvider`?) type to encapsulate this setup code, the distinction between key types and the available fields?